### ISSUE TITLE
Enhancement request for per-folder(device) sync conditions

### ISSUE THREAD

**Issue by Catfriend1** [Catfriend1/syncthing-android#110]
Quote from @cdrfun at ( https://github.com/Catfriend1/syncthing-android/issues/57#issuecomment-432148444 )
> Really nice work! 
I had some time to test today, and here is what I've found. (If needed I'll create a separate issue(s)). Some of my labels might not be correct, because I'm using the German version. I use numbers for better referencing, not to indicate any importance:
1. Folder-Sync condition Start at selected Wifi can't be activated when a option page for an folder is opened where start with wifi is activated. When deactivating/activating the option start with wifi, the option Start at selected Wifi can be set.
2. Currently folder sync conditions can only limit the folders sync conditions compared to the global conditions. I understand that it is good to have some kind of a global limit. But this puts the user in a tricky situation when he wants one folder to "always sync". He has to max out the global conditions and limit all folders bit the always sync folder. additionally he has to remember to limit all new folders, or he will use more traffic than he thought. Perhaps a separation of syncthing starting conditions and default folder conditions would be a solution.
3. The Folder-Sync condition dialog has a checkmark in the upper right. This is not necessary, because changes are saved when going back (either using the arrow in the upper left or the back button). Additionally when pressing the checkmark the folder options dialog is revealed and here is the trash icon at the same position. This is not consistent.
4. When wifi connection is lost, and some folders are configured to run on mobile, syncthing seems to restart completely instead of just pausing non mobile folders. When the wifi connection is regained, syncthing just unpauses the folders - which is correct.

**Comment by Catfriend1** [comment]
@cdrfun As stated in the other issue > added to my development plan. :)
v0.14.51.8 / 4171 should have point (1) fixed already. I did find another UI glitch that does not impact functionality if you enable "sync on wifi" and "wifi whitelist" was disabled before, the wifi SSIDs must be re-enabled manually by toggling each SSID to enabled. So far I didn't find a reproducer that leaves some switch greyed out wrongly.

**Comment by cdrfun** [comment]
In v0.14.51.8 / 4171 I can definitely reproduce (1). I'll have a look again when the next apk is released.

I'd like to suggest another enchantment request:

6. Make battery options available in Folder-Sync conditions.

**Comment by Catfriend1** [comment]
Ok noted. Thanks for checking again.

**Comment by Catfriend1** [comment]
@cdrfun How do you reproduce (1)?
Me:
![image](https://user-images.githubusercontent.com/16361913/47604211-92a0e400-d9f6-11e8-9f0d-927d6966a208.png)
=>
![image](https://user-images.githubusercontent.com/16361913/47604214-9896c500-d9f6-11e8-9c64-e626faddf44a.png)
=>
![image](https://user-images.githubusercontent.com/16361913/47604218-9cc2e280-d9f6-11e8-9e79-699b667d9c88.png)
=>
![image](https://user-images.githubusercontent.com/16361913/47604220-a2202d00-d9f6-11e8-9a6d-563c5c5b6267.png)
=>
![image](https://user-images.githubusercontent.com/16361913/47604225-ac422b80-d9f6-11e8-98a0-e2d74333f29c.png)

**Comment by Catfriend1** [comment]
Note: (3) was fixed in PR #114

**Comment by Catfriend1** [comment]
Long-term improvment noted: (2) + (4)

**Comment by Catfriend1** [comment]
New release containg the fixes for (1) + (3): v0.14.51.11

**Comment by cdrfun** [comment]
Great work 👍
Just to keep track I'll list the remaining issues:

2. Currently folder sync conditions can only limit the folders sync conditions compared to the global conditions. I understand that it is good to have some kind of a global limit. But this puts the user in a tricky situation when he wants one folder to "always sync". He has to max out the global conditions and limit all folders bit the always sync folder. additionally he has to remember to limit all new folders, or he will use more traffic than he thought. Perhaps a separation of syncthing starting conditions and default folder conditions would be a solution.
4. When wifi connection is lost, and some folders are configured to run on mobile, syncthing seems to restart completely instead of just pausing non mobile folders. When the wifi connection is regained, syncthing just unpauses the folders - which is correct.
5. Make battery options available in Folder-Sync conditions.

**Comment by Catfriend1** [comment]
Personal Dev note: I need to do this after refactoring ConfigXml module.

**Comment by Catfriend1** [comment]
So, I'm still on it figuring out a better implementation which can handle the per folder/device sync in a "whitelist thinking" manner. The first (great) piece of work is nearly done, as now, the wrapper is able to write the correct folder/device paused (or not paused) state into the config directly when syncthing is not running. This make it more streamlined as it before had to start the sync core and then quickly apply the paused/unpaused flag. Since we are now able to determine and set the right state BEFORE starting the sync core, we have a lot of possibilities to improve further.
In the second piece of work, I'll refactor the run conditions that they are evaluated and applied when the sync core is not running, so pausing and unpausing won't be triggered on the sync core startup but (as it should be) in RunConditionMonitor.
Stay tuned ...

**Comment by Catfriend1** [comment]
Related FR: "#145 Always make individual sync conditions UI available"

**Comment by Catfriend1** [comment]
I'm currently working on improving the clearness and "thinking positive" when it comes to the run conditions.

**Comment by Catfriend1** [comment]
@cdrfun : I've implemented another improvement, see [FR]"Only start syncthing native when at least one folder or device is/would be unpaused #189" which will help getting the last bit of your request done.

**Comment by Catfriend1** [comment]
We'll take another way forward and maybe do this later as said in PR:
"Only start syncthing native when at least one folder or device is/would be unpaused", PR #182

**Comment by Catfriend1** [comment]
@thumos333 -1
